src/transformers/models/bigscience176b/modeling_bigscience176b.py

```diff
@@ -262,46 +290,83 @@ def forward(
         # Output. [sq, b, h]
         # =================
 
-        output = self.dense(context_layer)
+        # aggregate results across tp ranks
+        slices = context_layer.shape[-1] / self.pretraining_tp
+        output_tensor = torch.zeros_like(context_layer)
+        for i in range(self.pretraining_tp):
+            output_tensor = output_tensor + F.linear(context_layer[:, :, int(i*slices):int((i+1)*slices)], self.dense.weight[:, int(i*slices):int((i+1)*slices)])
+        if not self.skip_bias_add:
+            output_tensor = output_tensor + self.dense.bias if self.dense.bias is not None else output_tensor
+            output_bias = None
+        else:
+            output_tensor = output_tensor
+            output_bias = self.dense.bias
+        output = output_tensor
+        # output = self.dense(context_layer)
 
         outputs = (output, present)
         if output_attentions:
             outputs += (None,) # TODO not implemented yet
 
-        return outputs  # a, present, (attentions)
+        return outputs, output_bias  # a, present, (attentions)
 
 
 # Copied from transformers.models.gpt2.modeling_gpt2.GPT2MLP with GPT2->BigScience176B
 class BigScience176BMLP(nn.Module):
-    def __init__(self, config):
+    def __init__(self, config, dtype=torch.bfloat16):
         super().__init__()
         hidden_size = config.hidden_size
-        self.dense_h_to_4h = nn.Linear(hidden_size, 4 * hidden_size)
-        self.activation_func = F.gelu
-        self.dense_4h_to_h = nn.Linear(4 * hidden_size, hidden_size)
+        self.skip_bias_add = config.skip_bias_add
+        self.pretraining_tp = config.pretraining_tp
+        self.dense_h_to_4h = nn.Linear(hidden_size, 4 * hidden_size, dtype=dtype)
+        # self.activation_func = F.gelu
+        self.activation_func = bias_gelu_impl
+        self.dense_4h_to_h = nn.Linear(4 * hidden_size, hidden_size, dtype=dtype)
 
     def forward(self, hidden_states):
-        hidden_states = self.dense_h_to_4h(hidden_states)
-        hidden_states = self.activation_func(hidden_states)  # TODO maybe we want to fuse the bias addition and gelu
-        hidden_states = self.dense_4h_to_h(hidden_states)
-        return hidden_states
+        input_ = hidden_states
+
+        hidden_states = self.activation_func(
+            F.linear(
+                hidden_states,
+                self.dense_h_to_4h.weight
+            ),
+            self.dense_h_to_4h.bias
+        )
+
+        intermediate_output = torch.zeros_like(input_)
+        slices = self.dense_4h_to_h.weight.shape[-1] / self.pretraining_tp
+        for i in range(self.pretraining_tp):
+            intermediate_output = intermediate_output + F.linear(hidden_states[:, :, int(i*slices):int((i+1)*slices)], self.dense_4h_to_h.weight[:, int(i*slices):int((i+1)*slices)])
```

Comment:
make sure you run `make style` or `make fixup` before merging in the main repo (we use code formating tools). Otherwise you'll have an error